scipy/sparse/_dok.py

```diff
@@ -368,23 +516,54 @@ def todok(self, copy=False):
     todok.__doc__ = _spbase.todok.__doc__
 
     def tocsc(self, copy=False):
+        if self.ndim == 1:
+            raise NotImplementedError("tocsr() not valid for 1d sparse array")
         return self.tocoo(copy=False).tocsc(copy=copy)
 
     tocsc.__doc__ = _spbase.tocsc.__doc__
 
+# why doesnt dok have one of these?
+#    def eliminate_zeros(self):
+#        self._dict = {k: v for k, v in self._dict.items() if v != 0}
+
     def resize(self, *shape):
-        shape = check_shape(shape)
+        is_array = isinstance(self, sparray)
+        shape = check_shape(shape, allow_1d=is_array)
+        if len(shape) != len(self.shape):
+            # TODO implement resize across dimensions
+            raise NotImplementedError
+
+        if self.ndim == 1:
```

Comment:
Recommend not implementing resize semantics; even with NumPy, its behavior is  inconsistent.